genericshomework/src/main/java/GenericBinaryTreeUtils.java

```diff
@@ -0,0 +1,80 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class GenericBinaryTreeUtils<T extends Comparable> {
+
+    private List<T> elementsSortedAscending = new ArrayList<T>();
+    private List<T> elementsSortedDescending = new ArrayList<T>();
+
+    /**
+     * Recursive method that traverses the given tree in order to produce an ascending sorted tree
+     *
+     * @param tree            The tree for which the sorted elements are requested
+     * @param receivedPointer Node type element
+     * @return an ascending sorted List of Nodes type objects
+     */
+    public List<T> getSortedTreeAscending(GenericBinaryTree tree, GenericBinaryTree.Node receivedPointer) {
```

Comment:
Why isn't this encapsulated in the binary tree? When you use TreeSet do you need a TreeSetUtils to see the sorted elements?